pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -223,8 +224,12 @@ protected void releaseAndRemoveAllSegments() {
       segmentDataManagers = new ArrayList<>(_segmentDataManagerMap.values());
       _segmentDataManagerMap.clear();
     }
-    for (SegmentDataManager segmentDataManager : segmentDataManagers) {
-      releaseSegment(segmentDataManager);
+    if (!segmentDataManagers.isEmpty()) {
+      ExecutorService stopExecutorService = Executors.newFixedThreadPool(segmentDataManagers.size());
+      for (SegmentDataManager segmentDataManager : segmentDataManagers) {
+        stopExecutorService.submit(() -> releaseSegment(segmentDataManager));
+      }
+      stopExecutorService.shutdown();
```

Comment:
do we need to await termination here?

Reply:
Good point! Added the code below to wait for at most 10 minutes before exiting the method.

Reply:
Don't we need to remove line 232 ?

Reply:
+1 to Sidd's comment here. We are calling shutdown here, and then adding a try catch to await termination for up to 10 minutes. based on the shutdown documentation:

```
    * Initiates an orderly shutdown in which previously submitted
     * tasks are executed, but no new tasks will be accepted.
     * Invocation has no additional effect if already shut down.
     *
     * <p>This method does not wait for previously submitted tasks to
     * complete execution.  Use {@link #awaitTermination awaitTermination}
     * to do that.
```